melodies_monet/plots/surfplots.py

```diff
@@ -880,6 +881,13 @@ def calculate_multi_boxplot(df, df_reg=None, region_name= None,column=None, labe
     plot_kwargs['label'] = label
     if df_reg is not None:
         comb_bx[label] = df_reg[column+'_reg']
+        siteid_bx['siteid']=df_reg['siteid']
+        region_list = []
+        for i in range(len(siteid_bx['siteid'])):
+            siteid_here = siteid_bx['siteid'][i]
+            region_here = df[df.siteid == siteid_here][region_name[0]].to_string().replace(' ',',').split(',')[-1]
+            region_list.append(region_here)
+        region_bx = pd.DataFrame(region_list,columns=['set_regions'])
```

Comment:
@zmoon This looks broadly good. Zach, there is probably an easier way to do this with either merge or join in pandas. The metadata is only in df and not in df_reg so we just need to add it into df_reg here by siteid. Can you advise Beiming on how to do this?

Reply:
@zmoon shall we merge this PR?

Reply:
I believe the 169facb change captured what we were looking for.

Reply:
Great, I agree.